app/models/alchemy/storage_adapter/active_storage/picture_url.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class StorageAdapter
+    # Returns the URL to a variant of a picture using ActiveStorage
+    class ActiveStorage::PictureUrl
+      attr_reader :picture, :image_file
+
+      # @param [Alchemy::Picture]
+      #
+      def initialize(picture)
+        @picture = picture
+        @image_file = picture.image_file
+      end
+
+      # The URL to a variant of a picture
+      #
+      # @return [String]
+      #
+      def call(options = {})
+        variant_options = DragonflyToImageProcessing.call(options)
+        variant_options[:format] = options[:format] || default_output_format
+        variant = image_file&.variant(variant_options)
+        return unless variant
+
+        Rails.application.routes.url_helpers.rails_blob_path(
```

Comment:
As discussed IRL, this will generate different URLs from the Dragonfly ones, and will render existing picture URLs invalid. Let's think about this a bit more.

Reply:
Maybe a controller or middleware that takes the dragonfly URL scheme and translates to activestorage?

Reply:
The dragonfly adapter is still the default adapter even with this change. Since we still need an upgrade path anyway, it is better dealt with in a separate PR.